nectar/src/rate.rs

```diff
@@ -114,6 +125,29 @@ impl Spread {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum RateRetrieval {
```

Comment:
A more scalable approach could be to have a trait for this and simply pass around a trait object like: `Box<GetRate>` or `&dyn GetRate`.

This allows you to _not_ have a single place in which all these different variants are mentioned. That in turn makes it easier to parallelize changes because it is less likely that changes to different modules conflict with each other. Another way of putting it would be to "optimize for deletion". Optimize the design in a way so that individual features can easily be deleted without interfering with other changes.